### PR TITLE
Update Scala Versions in .travis.yml

### PR DESCRIPTION
**Pull request by ChristopherDavenport**
With the release of Scala 2.12.2 and 2.11.10 these changes can be reflected in our builds.

I'm sure I'm missing a location where the primary value is being populated by an sbt file. Picked the furthest upstream branch to proliferate.

### CODE REVIEW

**Comment by ChristopherDavenport** [comment]
Interesting. Error is 
```
[info] core: found 2 potential binary incompatibilities while checking against org.http4s:http4s-core_2.12:0.15.0a 
[error]  * the type hierarchy of object org.http4s.package#HttpService is different in current version. Missing types {scala.Serializable,java.io.Serializable}
[error]    filter with: ProblemFilters.exclude[MissingTypesProblem]("org.http4s.package$HttpService$")
[error]  * the type hierarchy of object org.http4s.package#AuthedService is different in current version. Missing types {scala.Serializable,java.io.Serializable}
[error]    filter with: ProblemFilters.exclude[MissingTypesProblem]("org.http4s.package$AuthedService$")
```

**Comment by bfritz** [comment]
Deferring to @rossabaker on binary compatibility issues.  Might be worth bumping `scalaVersion` in `docs/src/main/tut/service.md`, but LGTM. :+1:

**Comment by bfritz** [comment]
Just realized 2.11.11 is the latest (and last) in the 2.11 series:
https://www.scala-lang.org/news/releases-1Q17.html

Should we upgrade to .11 instead of .10?

**Comment by ChristopherDavenport** [comment]
Woops, missed that 2.11.10 was an issue as well.

**Comment by rossabaker** [comment]
The MiMa issue looks similar to https://github.com/scala/scala-dev/issues/290.  These objects are awkwardly in a package object to work around a REPL bug that I'm no longer sure exists.  We could explicitly extend those types, and then look into getting them out of the package object in 0.16.

The Scala bumps won't work on 0.16 and above because those releases don't yet exist for Typelevel Scala.  Those corresponding releases are expected soon.

File: build.sbt [hunk 1]
```diff
@@ -556,7 +556,7 @@ lazy val commonSettings = Seq(
   },
   coursierVerbosity := 0,
   ivyLoggingLevel := UpdateLogging.Quiet // This doesn't seem to work? We see this in MiMa
-)
+) ++ xlint
```
Comment (rossabaker) [inline build.sbt]: I think line 492 is the problem.  You're adding the right xlint here, but `-Xlint` is already in the list.

**Pull request merged**